core/src/tile/tileManager.cpp

```diff
@@ -384,11 +388,24 @@ void TileManager::loadSubTasks(std::vector<std::shared_ptr<DataSource>>& _subSou
         if (it != subTasks.end() && (*it)->subTaskId() == int(index)) { continue; }
 
         TileID subTileID = tileID;
+
+        // get tile for lower zoom if we are past max zoom
         if (subTileID.z > subSource->maxZoom()) {
             subTileID = subTileID.withMaxSourceZoom(subSource->maxZoom());
         }
         auto subTask = subSource->createTask(subTileID, index);
-        if (subTask->isReady()) {
+        // check if we are at valid zoom for source
+        if (!subSource->isActiveForZoom((float)subTileID.z)) {
+            // Right now all subSources are raster, but let's check anyway...
+            if (RasterSource* rasterSubSource = dynamic_cast<RasterSource*>(subSource.get())) {
+                subTasks.insert(it, subTask);
+                // If the subSource isn't active for the zoom, we should just
+                // load an empty texture and move on.
+                rasterSubSource->loadEmptyTexture(std::move(subTask));
```

Comment:
The reason it won't render the uber time is because it is not "ready" yet. A main tile only becomes ready to be drawn is when all its dependent source tiles are loaded. 

Along with what Brett mentioned,  there is no point processing a master tile if it's subtIle will never be fetched,  we can ignore all such tiles and save on some processing.